site/viamguides/src/qrcode/qrcode.md

```diff
@@ -0,0 +1,131 @@
+1. In [the Viam app](https://app.viam.com/robots), find the **CONFIGURE** tab. It's time to configure your hardware.
+1. Click the **+** icon in the left-hand menu and select **Component**.
+   ![select component](assets/component.png)
+1. Select `camera`, and find the `webcam` module. This adds the module for working with a USB webcam. Leave the default name `camera-1` for now.
+   ![select webcam](assets/camera.png)
+1. Notice adding this component adds the webcam hardware component called `camera-1`. The collapsible panel on the right corresponds to the part listed in the left sidebar. From the **Attributes** section of the panel, select a `video_path`.
+   ![select video path](assets/videoPath.png)
+1. Click **Save** in the top right to save and apply your configuration changes.
+1. At the bottom of the `camera-1` panel, expand the **TEST** section to ensure you have configured the camera properly.
+   ![test camera](assets/test.png)
+   > aside negative
+   > If any problems occur, check under the **LOGS** tab to see what might be going wrong.
+
+<!-- ------------------------ -->
+
+## Add a vision service
+
+Duration: 3
+
+Now that your hardware is working the way you want it, it's time to add a vision service to detect and decode a QR code.
+
+1. In the Viam app, click the **+** icon in the left-hand menu and select **Service**, and then `vision`.
+   ![select vision](assets/vision.png)
+1. Search for a module called `pyzbar`. Then click **Add module**, and **Create** a new vision service called `vision-1`.
+   ![add module](assets/pyzbar.png)
+1. Notice this creates two new items in the left sidebar. The first is your new vision service called `vision-1`, and the second is your new `pyzbar` module.
+1. In the `vision-1` panel under the **Depends on** section, check the `camera-1` resource. This configures the vision service to depend on data coming in from the webcam.
+1. **Save** your changes in the top right and wait a few moments for the configuration changes to take effect.
+1. At the bottom of the `vision-1` panel, expand the **TEST** section to ensure you have configured the vision service properly. Point the webcam towards a QR code to see if the camera detects it.
+1. If the scanner detects a QR code, a bounding box will highlight the QR code in the video feed, and decode the data and display it on the right under **Labels**. In the example shown here, the URL decoded was `viam.com`.
+   ![qr code test](assets/qrCodeTest.png)
+
+   > aside negative
+   > **Troubleshooting:**
+   >
+   > - Check under the **LOGS** tab of the Viam app to see what might be going wrong.
+   > - Make sure you are testing with a valid QR code. You can generate QR codes [here](https://qrnobs.com/).
+   > - Make sure you are using a high-quality USB webcam to ensure accurate QR code detection
+
+<!-- ------------------------ -->
+
+## Next Steps
+
+Duration: 3
+
+Now that you have a camera that detects and decodes QR codes, you can do the following:
```

Comment:
Some interesting use cases for QR codes in automation projects that could be included as next steps:

- instructions/labels for roving bots that move around an office, home, or outdoor space
- security access to disable alarms or raise a barrier when the correctly encoded QR is shown
- reading the linked digital menus through a text-to-speech (TTS) module for people who can't view them normally due to vision impairment